src/DirectoryTree.py

```diff
@@ -0,0 +1,111 @@
+                next_path = os.path.abspath(os.path.join(dirname, entry))
+                if self.file_to_skip(next_path):
+                    continue
+                next_file = File(next_path)
+                child_size = os.path.getsize(next_path)
+                next_file.set_size(child_size)
+                dir_size += child_size
+                root.add_next_child(next_file)
+                files_dict[next_path] = next_file
+
+            for entry in subdirs:
+                next_path = os.path.abspath(os.path.join(dirname, entry))
+                if self.file_to_skip(next_path):
+                    continue
+                next_file = files_dict[next_path]
+                dir_size += next_file.get_size()
+                root.add_next_child(next_file)
+
+            root.set_size(dir_size)
+            for child in root.get_children():
+                child.set_parent_size(dir_size)
+            files_dict[dirname] = root
+        
+        files_dict[path].set_parent_size(0)
+        return files_dict[path], files_dict
+
+    def check_access(self, path):
+        return os.access(path, os.R_OK)
+
+    def file_to_skip(self, path):
+        return not self.check_access(path) or os.path.islink(path)
+
+    def get_parent(self, path):
+        return os.path.dirname(path) if os.path.isfile(path) else os.path.abspath(os.path.join(path, ".."))
+
+    def walklevel(self, path=''):
+        path = path if path != '' else self.path
+        if path.startswith(("~", "/")):
+            path = os.path.expanduser(path)
+
+        root = self.files[path]
+        for child in root.get_children():
+            yield child
+
+    def remove_file(self, path):
+        try:
+            if not self.check_access(path):
+                raise Exception('You have not access to {} or file does not exists'.format(path))
+            file_to_remove = self.files[path]
+            size_of_removed_file = file_to_remove.get_size()
+            keys = self.files.keys()
+            path_copy = str(path)
+            while path_copy != '/':
+                path_copy = os.path.abspath(os.path.join(path_copy, '..'))
+                if not path_copy in keys:
+                    break
+                self.files[path_copy].decrease_sizes(size_of_removed_file)
+            name = file_to_remove.get_name()
+            parent_path = os.path.abspath(os.path.join(path, '..'))
+            if parent_path in keys:
+                parent_dir = self.files[parent_path]        
+                for i, file in enumerate(parent_dir.get_children()):
+                    if file.get_name() == name:
+                        parent_dir.delete_child(i)
+            os.remove(path)
+        except KeyError:
+            raise KeyError('File {} does not exists'.format(path.split('/')[-1]))
+        except Exception as e:
+            raise Exception(e)
+        
+    def remove_directory(self, dir_path):
+        try:
+            if not self.check_access(dir_path):
```

Comment:
you've repeated 20 lines of code, try to extract it to another function
you can name it `remove_tree_node` or sth like that because I can see that all it does is traversing the tree upwards and correcting the sizes

Reply:
Done. Additionally, I divided this one method to smaller methods.